sdk/keyvault/Azure.Security.KeyVault.Secrets/src/SecretClient.cs

```diff
@@ -51,14 +51,17 @@ public SecretClient(Uri vaultUri, TokenCredential credential)
         /// <param name="vaultUri">Endpoint URL for the Azure Key Vault service.</param>
         /// <param name="credential">Represents a credential capable of providing an OAuth token.</param>
         /// <param name="options">Options that allow to configure the management of the request sent to Key Vault.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="vaultUri"/> or <paramref name="credential"/> is null.</exception>
         public SecretClient(Uri vaultUri, TokenCredential credential, SecretClientOptions options)
         {
-            _vaultUri = vaultUri ?? throw new ArgumentNullException(nameof(credential));
+            _vaultUri = vaultUri ?? throw new ArgumentNullException(nameof(vaultUri));
             options = options ?? new SecretClientOptions();
```

Comment:
Good catch.  We should also check credential here though, and remove the check from the internal class.